lib/driver.js

```diff
@@ -435,6 +442,43 @@ class AndroidDriver extends BaseDriver {
     }
   }
 
+  async setSharedPreferences () {
+    let sharedPrefs = this.opts.sharedPreferences;
+    log.info("Trying to set shared preferences");
+    let name = sharedPrefs.name;
+    if (_.isUndefined(name)) {
+      log.warn(`Skipping setting Shared preferences, name is undefined: ${JSON.stringify(sharedPrefs)}`);
+      return false;
+    }
+    let remotePath = `/data/data/${this.opts.appPackage}/shared_prefs`;
+    let remoteFile = `${remotePath}/${name}.xml`;
+    let localPath = `/tmp/${name}.xml`;
+    let builder = this.getPrefsBuilder();
+    builder.build(sharedPrefs.prefs);
+    log.info(`Creating temporary shared preferences: ${localPath}`);
+    builder.toFile(localPath);
+    log.info(`Creating shared_prefs remote folder: ${remotePath}`);
+    await this.adb.shell(['mkdir', '-p', remotePath]);
+    log.info(`Pushing shared_prefs to ${remoteFile}`);
+    await this.adb.push(localPath, remoteFile);
+    try {
+      log.info(`Trying to remove shared preferences temporary file`);
+      if (await fs.exists(localPath)) {
+        await fs.unlink(localPath);
+      }
+    } catch (e) {
+      log.warn(`Error trying to remove temporary file ${localPath}`);
+    }
+    return true;
+  }
+
+  getPrefsBuilder () {
```

Comment:
is there any reason to create and reuse this rather than just create a new instance when we call `setSharedPreferences`? don't we only call this one per session anyway?